docs/guides/testing-keycloak.rst

```diff
@@ -0,0 +1,111 @@
+Testing Keycloak integration
+############################
+
+Keycloak integration can be tested using the related methods provided in
+:class:`sssd_test_framework.roles.keycloak.Keycloak`.
+
+Multhost test configuration requirements
+----------------------------------------
+
+Testing with Keycloak integration requires additional systems configured to
+provide a Keycloak server and the ipa-tuura SCIMv2 **bridge** service:
+
+.. code-block:: yaml
+
+    domains:
+    - id: sssd
+      hosts:
+      - hostname: idm.ipa.test
+        role: ipa
+        ssh:
+          host: HOSTNAME_IPA
+          username: root
+          password: Secret123
+        config:
+          client:
+          ipa_domain: ipa.test
+
+      - hostname: bridge.ipa.test
+        role: client
+        ssh:
+          host: HOSTNAME_BRIDGE
+          username: root
+          password: Secret123
+
+      - hostname: keycloak.ipa.test
+        role: keycloak
+        ssh:
+          host: HOSTNAME_KEYCLOAK
+          username: root
+          password: Secret123
+        config:
+          keycloak:
+            keycloak_server: keycloak.ipa.test
+            keycloak_admin_user: admin
+            keycloak_admin_password: Secret123
+
+It should be noted that at this time, the ipa-tuura bridge service does not
```

Comment:
How about create ipa-tuura bridge role and just inherit the class from Client?

Reply:
My original thought was that an ipa-tuura bridge role wouldn't be needed since Keycloak handles most of the interaction with the service through the new storage plugin.   I had added this to use ssh to implement a workaround in a test case but, that has been fixed directly in ipa-tuura now so it's no longer needed.

I'll look into this more today though.

Reply:
So it may not be needed from API point of view, but you need the host available and the hosts must meet some conditions (SCIMv2 server installed, etc). These conditions are described by the role. So even if you don't access this host, you want to add a specific role to it so you can say that this test requires "keycloak, ipa and ipa-tuura bridge". Now you don't require ipa-tuura bridge and if I pass client container it will still run the test.

If you just put client there, it makes it something different. The requirements are formally described here: https://tests.sssd.io/en/latest/config.html

Reply:
ok, I'll add an ipa-tuura bridge role.